### PR TITLE
Add support for filtering file names for Javadoc plugin.

### LINKED ISSUE #1058: Javadoc maven plugin includes all Java files
Add support for include and exclude for java files (based on file name).

### PR DESCRIPTION
**Pull request by tomas-langer**
Resolves #1058

### CODE REVIEW

File: maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java [hunk 1]
```diff
@@ -750,8 +768,12 @@ private Set<SourceRoot> sourceRootsFromSourceJar(MavenArtifact dep) {
 
     private Set<SourceRoot> sourceRootsFromProjectFiles(Path dir) {
         List<Path> nested = Lists.filter(workspace.values(), it -> !it.equals(dir) && it.startsWith(dir));
+        DirectoryScanner ds = new DirectoryScanner();
```
Comment (romain-grecourt) [inline maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java:771]: The DirectoryScanner seems unused.

File: maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java [hunk 2]
```diff
@@ -236,6 +237,20 @@ public class JavadocMojo extends AbstractMojo {
     @Parameter(property = "helidon.javadoc.pomExcludes")
     private List<String> pomExcludes = List.of();
 
+    /**
+     * File name patterns.
+     * List of include file names.
+     */
+    @Parameter(property = "helidon.javadoc.fileIncludes", defaultValue = "*.java")
+    private List<String> fileNameIncludes = List.of();
```
Comment (romain-grecourt) [inline maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java:245]: Can we use the same name, in configuration and property.
I like `fileIncludes` over `fileNameIncludes`.

**Review by romain-grecourt — COMMENTED** [review]
Can you update the IT test to use the new option.

E.g. You could add a new .java file that is excluded with a fileExclude configuration, and update ProjectsTestIT.test1 to assert that the generated javadoc does not exist.

File: maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java (hunk 1 above)
Comment (tomas-langer) [reply maven-plugins/javadoc-maven-plugin/src/main/java/io/helidon/build/javadoc/JavadocMojo.java:771]: yes, discovered the `MavenFilters` class that does what I needed.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**